src/output_modules/module_json.c

```diff
@@ -50,7 +50,7 @@ int json_output_file_init(struct state_conf *conf, UNUSED char **fields,
 	return EXIT_SUCCESS;
 }
 
-char *hex_encode(char *packet, int buflen)
+char *hex_encode(unsigned char *packet, int buflen)
```

Comment:
The hex_encode functions in all other output modules follow this format (which is why I used it). I actually think forcing it unsigned is the wrong answer. You should be using %.2hhx which makes the signage irreverent.